### PR TITLE
Create the input component for weight

### LINKED ISSUE #62: Create the input component for weight
ℹ️ This issue is part of Epic #59 

## Objective

Create the input component for entering the weight according to the design: [Figma ](https://www.figma.com/file/2SHaHIlfaR031SLnJPXgIR/OKdose?node-id=71%3A3498&t=vQ9mmEW7CYmzH4Zr-0)

image
image
image

## Checklist

- [ ] There is a title that is provided as a prop when using the component.
- [ ] There is a description that is provided as a prop when using the component.
- [ ] There is an input to enter the weight with a placeholder as shown in the design
- [ ] There are upper and lower bounds as props in the component
- [ ] It is displayed an alert card if the weight is not within the limits and the input text changes the color to red as can be shown in the image above. 
- [ ] There is a button common component with the button text as a prop when using the component with the possibility of being disabled

### PR DESCRIPTION
**Pull request by keliumJU**
## Description

Add the input component for entering the weight according to the design

image

image

image

Resolves #62 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: okdose-web/assets/locales/en/translation.json [hunk 1]
```diff
@@ -2,11 +2,17 @@
   "app_info": {
     "weight_button": "Calculate",
     "back_button": "Back",
-    "dosage": "Dosage",
```
Comment (jcamilocp) [inline okdose-web/assets/locales/en/translation.json:5]: Are we not using this key anywhere else?

File: okdose-web/components/CardInputWeight.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import Button from './common/Button';
+import {useEffect, useState} from 'react';
+import AlertCard from './common/AlertCard';
+import arrowRightIcon from '../assets/images/icons/arrowRightIcon.svg';
```
Comment (jcamilocp) [inline okdose-web/components/CardInputWeight.js:4]: Should we declare shorter routes for icons?

File: okdose-web/components/CardInputWeight.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  description,
+  showCategoryIcon,
+  showTitleCard,
+  titleCard,
+  upperBound,
+  lowerBound,
+  disableComponent
+}) {
+  const [inputValue, setInputValue] = useState();
+  const [fontStyle, setFonStyle] = useState('');
```
Comment (jcamilocp) [inline okdose-web/components/CardInputWeight.js:19]: A typo here, `setFonStyle`.

File: okdose-web/components/CardInputWeight.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  function handleChange (event) {
+    let value = event.target.value;
+    setInputValue(parseInt(value));
+    if (value < lowerBound || value > upperBound) {
+      setFonStyle('error');
+    } else {
+      setFonStyle('normal');
+    }
+  }
```
Comment (jcamilocp) [inline okdose-web/components/CardInputWeight.js:53]: Let's put a newline here

File: okdose-web/components/common/AlertCard.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function AlertCard ({msg}) {
+  return (
+    <p className='mb-10 bg-[#FFD9D9] text-[#DC2626] text-sm font-normal'>
+      {msg}
+    </p>
+  );
+}
```
Comment (jcamilocp) [inline okdose-web/components/common/AlertCard.js:7]: Newline here

File: okdose-web/components/common/AlertCard.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+export default AlertCard;
```
Comment (jcamilocp) [inline okdose-web/components/common/AlertCard.js:8]: Don't we need prop types here?

File: okdose-web/components/common/Button.js [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
-    back: 'w-[185px] h-[50px] bg-sky-500 rounded-lg',
-    weight: 'w-[395px] h-[50px] bg-indigo-600 rounded'
+    back: 'w-1/3 h-[50px] bg-sky-500 rounded-lg',
+    weight: 'w-full h-[50px] bg-indigo-600 rounded shadow',
+    weightDisable: 'w-full h-[50px] bg-[#CED2DE] rounded shadow'
   };
 
   return (
     <button
+      disabled={disable}
```
Comment (jcamilocp) [inline okdose-web/components/common/Button.js:13]: Maybe we should name this prop "disabled".

File: okdose/package.json [hunk 8]
```diff
@@ -21,7 +21,6 @@
     "eslint-plugin-jest": "^27.2.1",
     "eslint-plugin-no-loops": "^0.3.0",
     "jest": "^29.4.1",
-    "jest-svg-transformer": "^1.0.0",
```
Comment (jcamilocp) [inline okdose/package.json:24]: Is the `okdose/yarn.lock` file updated with these changes?

**Review by jcamilocp — CHANGES_REQUESTED** [review]
Hi @keliumJU, I left some comments here

File: okdose-web/assets/locales/en/translation.json (hunk 1 above)
Comment (keliumJU) [reply okdose-web/assets/locales/en/translation.json:5]: Probably not used elsewhere, it is part of the cardInputWeight title.
image

File: okdose-web/components/common/AlertCard.js (hunk 6 above)
Comment (keliumJU) [reply okdose-web/components/common/AlertCard.js:8]: or course I did, I forgot

File: okdose/package.json (hunk 8 above)
Comment (keliumJU) [reply okdose/package.json:24]: now if it is updated

File: okdose-web/components/CardInputWeight.js (hunk 2 above)
Comment (keliumJU) [reply okdose-web/components/CardInputWeight.js:4]: I improved it with the module resolver plugin

File: okdose-web/components/CardInputWeight.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+      setButtonStyle('weightDisabled');
+      setFontStyle('disable');
+    } else {
+      setButtonStyle('weight');
+      setFontStyle('normal');
+    }
+  }, []);
+
+  const inputStyle = {
+    error: 'text-[#DC2626] placeholder-[#DC2626]',
```
Comment (jcamilocp) [inline okdose-web/components/CardInputWeight.js:33]: Let's define this color as a custom color, so it's more organized

File: okdose-web/components/common/AlertCard.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+function AlertCard ({msg}) {
+  return (
+    <p className='mb-10 bg-[#FFD9D9] text-[#DC2626] text-sm font-normal'>
```
Comment (jcamilocp) [inline okdose-web/components/common/AlertCard.js:5]: Actually, let's define all colors like these as custom colors

File: okdose-web/components/common/Button.js [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
 import PropTypes from 'prop-types';
 
-function Button ({type, title}) {
+function Button ({type, title, disabled}) {
   const buttonStyles = {
-    back: 'w-[185px] h-[50px] bg-sky-500 rounded-lg',
-    weight: 'w-[395px] h-[50px] bg-indigo-600 rounded'
+    back: 'w-1/3 h-[50px] bg-sky-500 rounded-lg',
+    weight: 'w-full h-[50px] bg-indigo-600 rounded shadow',
+    weightdisabled: 'w-full h-[50px] bg-[#CED2DE] rounded shadow'
```
Comment (jcamilocp) [inline okdose-web/components/common/Button.js:8]: Another candidate for custom color here

File: okdose-web/jsconfig.json [hunk 12]
```diff
@@ -2,12 +2,8 @@
   "compilerOptions": {
     "baseUrl": ".",
     "paths": {
-      "@leishmaniasis": [
-        "../okdose/transmission-types/transmitted-by-vectors/leishmaniasis"
-      ],
-      "@nearInterface": ["../okdose/near/nearInterface"],
```
Comment (jcamilocp) [inline okdose-web/jsconfig.json:8]: Can we have an explanation for these changes?

**Review by jcamilocp — CHANGES_REQUESTED** [review]
Let's set some custom colors for the whole app. We can name them by their function if needed (for example, red-error-text).

Also, I left a comment for some odd changes, can we have a little message describing the changes and why did we do those? Thanks!

File: okdose-web/jsconfig.json (hunk 12 above)
Comment (keliumJU) [reply okdose-web/jsconfig.json:8]: Moving the folder from **okdose/near** to **okdose-web/near** changes its path, also they were created at the beginning of the project for testing purposes, so now they are not being used, that's why I remove them, the module resolver is something that we activate now with the icons.

File: okdose-web/tailwind.config.js [hunk 13]
```diff
@@ -9,7 +9,11 @@ module.exports = {
     extend: {
       colors: {
         'gray-welcome': '#838383',
-        'blue-pale': '#BFDBFE'
+        'blue-pale': '#BFDBFE',
```
Comment (jcamilocp) [inline okdose-web/tailwind.config.js:12]: Can we rename this to `light-blue`? To preserve the same standard

**Review by jcamilocp — CHANGES_REQUESTED** [review]
Last very small change

File: okdose-web/tailwind.config.js (hunk 13 above)
Comment (keliumJU) [reply okdose-web/tailwind.config.js:12]: I think they have many variations, I'm going to change it to `light-blue` and gray-welcome to `soft-gray`

**Review by jcamilocp — APPROVED** [review]
LGTM! 👍

**Pull request merged**